tests/example_apps/mega/tables.py

```diff
@@ -22,37 +22,73 @@
     Timestamptz,
     Varchar,
 )
+from piccolo.engine.finder import engine_finder
 from piccolo.table import Table
 
+engine = engine_finder()
+
 
 class SmallTable(Table):
     varchar_col = Varchar()
 
 
-class MegaTable(Table):
-    """
-    A table containing all of the column types, and different column kwargs.
-    """
-
-    bigint_col = BigInt()
-    boolean_col = Boolean()
-    bytea_col = Bytea()
-    date_col = Date()
-    foreignkey_col = ForeignKey(SmallTable)
-    integer_col = Integer()
-    interval_col = Interval()
-    json_col = JSON()
-    jsonb_col = JSONB()
-    numeric_col = Numeric(digits=(5, 2))
-    real_col = Real()
-    double_precision_col = DoublePrecision()
-    smallint_col = SmallInt()
-    text_col = Text()
-    timestamp_col = Timestamp()
-    timestamptz_col = Timestamptz()
-    uuid_col = UUID()
-    varchar_col = Varchar()
+if engine.engine_type != "cockroach":  # type: ignore
+
+    class MegaTable(Table):  # type: ignore
+        """
+        A table containing all of the column types, different column kwargs.
+        """
+
+        bigint_col = BigInt()
+        boolean_col = Boolean()
+        bytea_col = Bytea()
+        date_col = Date()
+        foreignkey_col = ForeignKey(SmallTable)
+        integer_col = Integer()
+        interval_col = Interval()
+        json_col = JSON()
+        jsonb_col = JSONB()
+        numeric_col = Numeric(digits=(5, 2))
+        real_col = Real()
+        double_precision_col = DoublePrecision()
+        smallint_col = SmallInt()
+        text_col = Text()
+        timestamp_col = Timestamp()
+        timestamptz_col = Timestamptz()
+        uuid_col = UUID()
+        varchar_col = Varchar()
+
+        unique_col = Varchar(unique=True)
+        null_col = Varchar(null=True)
+        not_null_col = Varchar(null=False)
+
+else:
+
+    class MegaTable(Table):  # type: ignore
+        """
+        Special version for Cockroach.
+        A table containing all of the column types, different column kwargs.
+        """
+
+        bigint_col = BigInt()
+        boolean_col = Boolean()
+        bytea_col = Bytea()
+        date_col = Date()
+        foreignkey_col = ForeignKey(SmallTable)
+        integer_col = BigInt()
+        interval_col = Interval()
+        json_col = JSONB()
+        jsonb_col = JSONB()
+        numeric_col = Numeric(digits=(5, 2))
+        real_col = Real()
+        double_precision_col = DoublePrecision()
+        smallint_col = SmallInt()
+        text_col = Text()
+        timestamp_col = Timestamp()
+        timestamptz_col = Timestamptz()
+        uuid_col = UUID()
+        varchar_col = Varchar()
 
-    unique_col = Varchar(unique=True)
-    null_col = Varchar(null=True)
-    not_null_col = Varchar(null=False)
+        unique_col = Varchar(unique=True)
+        null_col = Varchar(null=True)
+        not_null_col = Varchar(null=False)
```

Comment:
Are these the same, except for the `JSON` column? As `JSON.column_type` returns `JSONB` for Cockroach, should it still work without needing the modified version of `MegaTable`?

Reply:
Yup, we need the special table here.

The test expects a `JSON` to be returned for `test_generate.py`. We can never know if what we sent to Cockroach was `JSON` or `JSONB`. Cockroach will always convert to `JSONB`.

The developer can use `JSON` or `JSONB`, it does not matter. This test is for integrity behind the scenes.

Reply:
Added docs: https://github.com/piccolo-orm/piccolo/pull/608/commits/ebe06e7e0286e4c2484bbd626740435eb50297f9